src/LeanMapper/DefaultMapper.php

```diff
@@ -30,6 +30,18 @@ class DefaultMapper implements IMapper
 
 
     /**
+     * @param  string|null
+     */
+    public function __construct($defaultEntityNamespace = 'Model\Entity')
+    {
+        if ($this->defaultEntityNamespace === 'Model\Entity') { // because BC
+            $this->defaultEntityNamespace = $defaultEntityNamespace;
```

Comment:
Co že má tohle dělat?

Reply:
Potřebujeme zajistit, aby to bylo kompatibilní s tímto kódem:

``` php
class MyMapper extends LeanMapper\DefaultMapper
{
    protected $defaultEntityNamespace = 'App\Entities';
}
```

Kdyby tam ta podmínka nebyla, přepíše konstruktor uživatelem nastavenou hodnotu property. Ta podmínka by tedy měla zaručit to, že k přepsání dojde jen v případě, že nebyla property změněna na jinou hodnotu.

Alternativně by šlo asi použít i následující kód:

``` php
public function __construct($defaultEntityNamespace = FALSE) {
    $this->defaultEntityNamespace = $defaultEntityNamespace !== FALSE ? $defaultEntityNamespace : $this->defaultEntityNamespace;
}
```

Ale oboje mi přijde podobně špatné :)

Reply:
Imho nejjednodušší by bylo tomu dát defaultní hodnotu na `null` a pokud je nenulová, nastavit hodnotu.
BC je zajištěná tím, že ti, kdo ten konstruktor doposud nepoužívali v něm budou mít null a proto zůstane nastavená defaultní, respektive překrytá hodnota.

Reply:
@Thoronir42 to sice ano, ale pak by nefungovalo volání `new LeanMapper\DefaultMapper(NULL)` - místo `NULL` by se to nastavilo na defaultní hodnotu `Model\Entity`. Pravděpodobně ale celý ten fallback odstraním, u nové velké verze by drobný BC break neměl vadit.

Reply:
Aaaaa.... Tahle motanice bude asi souviset s prázdným namespacem entit? Na to by vzhledem k BC pořád mohlo fungovat nechat tam uživatele poslat prázdnej string. Imho ta property by neměla v phpDocu připouštět null vůbec.

Potom by logika namespace entit v konstruktoru mohla vypadat asi takhle:
- pokud `is_null($defaultEntityNamespace)`, nech propertu - pro aktuální uivatele LM
- (pokud `!is_string($defaultEntityNamespace)`, **křič**)
- jinak nastav $defaultEntityNamespace.

Takhle mi to přijde dost čistě s ohledem na BC a nadcházející striktní typovost